website/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx

```diff
@@ -0,0 +1,314 @@
+Any property you add to your application configuration file under the prop [`additionalEnv`](/api-reference/application-config#additionalenv) will also be available in the `context.environment` object.
+
+Let's say you have something like this `custom-application-config.mjs` file in your project:
+
+```json
+{
+  "name": "My Custom Application",
+  "entryPointUriPath": "my-custom-application",
+  "cloudIdentifier": "gcp-eu",
+  "additionalEnv": {
+    "trackingSentry": "https://000@sentry.io/000",
+  },
+  ... // rest of your configuration
+}
+```
+
+Your `context.environment` object will look like this:
+
+```json
+{
+  "environment": {
+    "applicationId": "ckw3vt1hv034901uzio4bkzc3:my-custom-application",
+    "applicationName": "My Custom Application",
+    "entryPointUriPath": "my-custom-application",
+    "trackingSentry": "https://000@sentry.io/000",
+  }
+}
+```
+
+<Info>
+
+More information about accessing your custom configuration can found [here](/api-reference/application-config#runtime-application-environment)
+
+</Info>
+
+## Usage
+
+### useApplicationContext
+
+This is a custom hook which allows to access the `ApplicationContext` from any component (or other hooks).
+
+```jsx
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+
+const DisplayLocale = () => {
+  const { locale } = useApplicationContext((context) => ({
+    locale: applicationContext.dataLocale
+  }));
+  render() {
+    return (
+      <div>
+        <h1>Current locale: {locale}</h1>
+      </div>
+    );
+  }
+}
+
+export default DisplayLocale;
+```
+
+### withApplicationContext
+
+This is a High Order Component which allows to access the `ApplicationContext` from any component.
+It is meant to be used if you have a class based component where you need to read context data.
+
+```jsx
+import { withApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+
+class DisplayLocale extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Current locale: {this.props.locale}</h1>
+      </div>
+    );
+  }
+}
+
+export default withApplicationContext((applicationContext) => ({
+  locale: applicationContext.dataLocale
+}))(DisplayLocale);
+```
+
+# Project image settings
+
+When using your images for your projects you are expected to upload them to our platform. However, you can also choose to host your images in another service.<br/>
+If this is the case, we don't show them by default to prevent for huge images to be loaded in the browser that could impact Merchat Center performance.
+
+To enable the rendering of those images you need to set up a project-level configuration where you specify rewrite rules for different images sizes.<br/>
+You can find this configuration in the settings section of your projects in the Merchant Center (**Settings > Project Settings > Miscellaneous**).
```

Comment:
```suggestion
Product images can be uploaded to the commercetools platform or can be referenced from external sources.

By default, images referenced from external sources are not displayed in the Merchant Center. This is primarily due to avoid possible performance issues in case the size of those images is big.

In order to show images referenced from external sources, you need to define a configuration to rewrite the URL of the images to versions of the image that has a smaller size. You can configure that in the Merchant Center **Settings > Project Settings > Miscellaneous**.

This configuration can be fetched from your Custom Application using the following components.
```